indico/modules/events/editing/client/js/editing/EditableSubmissionButton.jsx

```diff
@@ -51,47 +54,74 @@ export default function EditableSubmissionButton({
+            onClose={() => setCurrentType(null)}
             closeIcon
             closeOnDimmerClick={false}
             closeOnEscape={false}
           >
             <Modal.Header>
-              {type === 'paper' && <Translate>Submit your paper</Translate>}
-              {type === 'slides' && <Translate>Submit your slides</Translate>}
+              {currentType === EditableType.paper && <Translate>Submit your paper</Translate>}
+              {currentType === EditableType.slides && <Translate>Submit your slides</Translate>}
+              {currentType === EditableType.poster && <Translate>Submit your poster</Translate>}
             </Modal.Header>
             <Modal.Content>
-              <Form id="submit-editable-form" onSubmit={handleSubmit}>
-                <FinalFileManager
-                  name="files"
-                  fileTypes={fileTypes}
-                  uploadURL={uploadURL}
-                  mustChange
-                />
-              </Form>
+              {currentType !== null && (
+                <Form id="submit-editable-form" onSubmit={handleSubmit}>
+                  <FinalFileManager
+                    name="files"
+                    fileTypes={getFileTypes({
+                      staticData: {
+                        fileTypes: fileTypes[currentType],
+                        contributionCode,
+                      },
+                    })}
+                    uploadURL={apiUploadURL({
+                      confId: eventId,
+                      contrib_id: contributionId,
+                      type: currentType,
+                    })}
+                    mustChange
+                  />
+                </Form>
+              )}
             </Modal.Content>
             <Modal.Actions style={{display: 'flex', justifyContent: 'flex-end'}}>
               <FinalSubmitButton form="submit-editable-form" label={Translate.string('Submit')} />
-              <Button onClick={() => setOpen(false)}>
+              <Button onClick={() => setCurrentType(null)}>
                 <Translate>Cancel</Translate>
               </Button>
             </Modal.Actions>
           </Modal>
         )}
       </FinalForm>
-      <button type="submit" className="i-button highlight" onClick={() => setOpen(true)}>
-        <Translate>Submit Files</Translate>
-      </button>
+      {editableTypes.length === 1 ? (
+        <Button onClick={() => setCurrentType(editableTypes[0])} primary>
+          <Translate>
+            Submit <Param name="editableType" value={editableTypes[0]} />
+          </Translate>
+        </Button>
+      ) : (
+        <Dropdown
+          className="primary"
+          text={Translate.string('Submit files')}
+          onChange={(__, {value}) => setCurrentType(value)}
+          options={editableTypes.map(editableType => ({
+            text: EditableTypeTitles[editableType],
+            value: editableType,
+            onClick: () => setCurrentType(editableType),
```

Comment:
do we still need this one? I think the `onChange` handler is enough

Reply:
Nope.